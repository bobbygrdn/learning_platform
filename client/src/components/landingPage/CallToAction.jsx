import React from 'react';
import '../../styles/landingPage/CallToAction.css';

export default function CallToAction() {
    return (
        <div className='callToAction' data-testid="callToAction">
            <p className='subheading' data-testid="subheading">Choose your path to eternal glory and fame</p>
            <p className='subheading' data-testid="subheading">Take on challenges, level up your knowledge and become a code warrior legend</p>
            <p className='subheading' data-testid="subheading">Discover a better way to become triumphant in your coding journey</p>
        </div>
    )
}
