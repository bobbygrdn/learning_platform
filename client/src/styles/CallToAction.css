.callToAction {
    display: flex;
    flex-direction: column;
    align-items: center;
    pointer-events: none;
    text-align: center;
}

.subheading {
    margin-bottom: 1em;
}

@media screen and (max-width: 600px) {
    .callToAction :nth-child(1) {
        font-size: 14px;
    }

    .callToAction :nth-child(2) {
        font-size: 14px;
        padding: 0em 1em;
    }

    .callToAction :nth-child(3) {
        font-size: 14px;
        padding: 0em 1em;
    }
}