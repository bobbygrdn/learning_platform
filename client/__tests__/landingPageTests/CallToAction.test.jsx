import React from 'react';
import { render, screen } from '@testing-library/react';
import CallToAction from '../../src/components/landingPage/CallToAction';

jest.mock('../../src/styles/landingPage/CallToAction.css', () => ({}));
jest.mock('../../src/styles/App.css', () => ({}));

it('renders without crashing', () => {
    const { container } = render(<CallToAction />);
    expect(container).toMatchSnapshot();
});

it('should render paragraph elements', () => {
    render(<CallToAction />);
    const paragraphs = screen.getAllByTestId('subheading');
    expect(paragraphs).toHaveLength(3);
});
