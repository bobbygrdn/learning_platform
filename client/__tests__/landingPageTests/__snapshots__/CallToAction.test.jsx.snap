// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="callToAction"
    data-testid="callToAction"
  >
    <p
      class="subheading"
      data-testid="subheading"
    >
      Choose your path to eternal glory and fame
    </p>
    <p
      class="subheading"
      data-testid="subheading"
    >
      Take on challenges, level up your knowledge and become a code warrior legend
    </p>
    <p
      class="subheading"
      data-testid="subheading"
    >
      Discover a better way to become triumphant in your coding journey
    </p>
  </div>
</div>
`;
